### PR TITLE
EOS-25060: Update cluster.conf to iem.conf

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

# Problem Statement
- https://jts.seagate.com/browse/EOS-25060

# Design
-  Create module specific configurations

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/setup/utils.py [hunk 1]
```diff
@@ -131,25 +133,27 @@ def _copy_cluster_map(conf_url_index: str):
         Conf.save('cluster')
 
     @staticmethod
-    def _create_cluster_config(server_info: dict):
+    def _create_iem_config(server_info: dict, machine_id: str):
         """ Create the config file required for Event Message """
-        for key, value in server_info.items():
-            Conf.set('cluster', f'node>{key}', value)
-        Conf.set('cluster', 'site_id', '1')
```
Comment (selvakumaar5496) [inline py-utils/src/setup/utils.py:138]: directly setting it to 1 does not make any sense

File: py-utils/src/setup/utils.py [hunk 2]
```diff
@@ -131,25 +133,27 @@ def _copy_cluster_map(conf_url_index: str):
...
-    def _copy_conf_sample_to_conf():
-        if not os.path.exists("/etc/cortx/cluster.conf.sample"):
-            with open("/etc/cortx/cluster.conf.sample", "w+") as file:
-                json.dump({}, file, indent=2)
-        # copy this sample conf file as cluster.conf
+        iem_index = 'iem'
+        with open('/etc/cortx/utils/iem.conf.sample', 'w+') as file:
+            json.dump({}, file, indent=2)
+        for _id in ['site_id', 'rack_id']:
+            if _id not in server_info.keys():
```
Comment (selvakumaar5496) [inline py-utils/src/setup/utils.py:142]: Check if it is available in ConfStore keys. (LR)
If not set it to 1. (LC)

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by rahul-tripathi-git — APPROVED** [review]
Looks good to me.

File: py-utils/src/setup/utils.py [hunk 3]
```diff
@@ -131,25 +133,28 @@ def _copy_cluster_map(conf_url_index: str):
...
-        Conf.set('cluster', 'site_id', '1')
-        Conf.set('cluster', 'rack_id', '1')
-        Conf.save('cluster')
+        iem_index = 'iem'
+        with open('/etc/cortx/utils/iem.conf.sample', 'w+') as file:
+            json.dump({}, file, indent=2)
+        for _id in ['site_id', 'rack_id']:
+            if _id not in server_info.keys():
+                server_info[_id] = 1
+        Conf.load(iem_index, 'yaml:///etc/cortx/utils/iem.conf.sample', \
```
Comment (sachinpunadikar) [inline py-utils/src/setup/utils.py:144]: this should be json .. rt ?

Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py]: yes. Done the change

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**